Core/src/Geometry/TrackingVolume.cpp

```diff
@@ -722,11 +715,7 @@ Acts::TrackingVolume::compatibleSurfacesFromHierarchy(
   }
 
   // Sort according to the path length
-  if (options.navDir == Direction::Forward) {
-    std::sort(sIntersections.begin(), sIntersections.end());
-  } else {
-    std::sort(sIntersections.begin(), sIntersections.end(), std::greater<>());
-  }
+  std::sort(sIntersections.begin(), sIntersections.end());
```

Comment:
the sorting doesn't change here, right?

Reply:
yes - the idea is that you always point the navigation code towards where you want to navigate and skip the forward backward checking. the sign of forward/backward is absorbed in the `direction` and distances are signed according to that direction

Reply:
nice